default/python/AI/AIstate.py

```diff
@@ -43,10 +43,62 @@
 outpostSystemIDs = []
 
 
-# AIstate class
+class ConversionError(Exception):
+    """Exception to be raised if the conversion of a savegame state fails.
+
+    Automatically logs and chats to the host if raised.
+    """
+    def __init__(self, msg=""):
+        print_error(msg)
+
+
+def convert_to_version(state, version):
+    """Convert a savegame AIstate to the next version.
+
+    :param state: savegame state, modified in function
+    :type state: dict
+    :param version: Version to convert to
+    :type version: int
+    """
+    print "Trying to convert savegame state to version %d..." % version
+    current_version = state.get("version", -1)
+    print "  Current version: %d" % current_version
+    if current_version == version:
+        raise ConversionError("Can't convert AI Savegame to the same version.")
+
+    if current_version > version:
+        raise ConversionError("Can't convert to an older version.")
+
+    if version != current_version + 1:
+        raise ConversionError("Can't skip a version.")
+
+    if version == 0:
+        pass  # only version number added
+    # elif version == 1:
+    #   state["some_new_member"] = some_default_value
+    #   del state["some_removed_member"]
+    #   state["list_changed_to_set"] = set(state["list_changed_to_set"])
+
+    print "  All updates set. Setting new version number."
+    state["version"] = version
+
+
 class AIstate(object):
-    """Stores AI game state."""
+    """Stores AI game state.
+
+    IMPORTANT:
+    If class members are redefined, added or deleted, then the
+    version number must be increased by 1 and the convert_to_version()
+    function must be updated so a saved state from the previous version
+    is playable with this AIstate version, i.e. new members must be added
+    and outdated members must be modified and / or deleted.
+    """
+    version = 0
+
     def __init__(self, aggression):
+        # Do not allow to create AIstate instances with an invalid version number.
+        assert(hasattr(AIstate, 'version') and isinstance(AIstate.version, int) and AIstate.version >= 0)
```

Comment:
So far, the only way I have been able to think of this assertion failing is if someone were to customize their AI code but mistakenly set the version to something like "3b" or whatnot.  Testing that scenario out, I just get the rather unhelpful log error and corresponding chat message 
```Chat Message to human: AI script Error in "startNewGame"```
And then when trying to advance to turn 2 the game would not advance (server seems stuck waiting for AI orders that will never arrive).

So, for that scenario, and any other scenarios you are trying to cover here, we need a smoother handling.   I would be inclined to handle all such scenarios by raising a ConversionError from the convert_to_version() function rather than any assertion here.